src/interpreter/runtime/castIntoStackLocalOfType.js

```diff
@@ -9,15 +9,26 @@ const f64 = require("./values/f64");
 export function castIntoStackLocalOfType(
   type: string,
   v: any,
-  nan: boolean = false
+  nan: boolean = false,
+  inf: boolean = false
 ): StackLocal {
   const castFn = {
     i32: i32.createValueFromAST,
     i64: i64.createValueFromAST,
-    f32: nan ? f32.createNanFromAST : f32.createValueFromAST,
+    f32: f32.createValueFromAST,
     f64: f64.createValueFromAST
   };
 
+  if (nan) {
```

Comment:
Could you please check for `nan === true and same for inf, it's a perf gain

Reply:
Ah cool! Any references on that? Ill do it before merging

Reply:
Yes it's bascially the ToBoolean algorythm which is called here to coerce `nan` to a boolean, if you do `nan === true` it doesn't need to.

Reply:
Makes sense, thanks!